itest/supply_commit_test.go

```diff
@@ -0,0 +1,134 @@
+package itest
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/taproot-assets/fn"
+	"github.com/lightninglabs/taproot-assets/tapgarden"
+	"github.com/lightninglabs/taproot-assets/taprpc"
+	"github.com/lightninglabs/taproot-assets/taprpc/mintrpc"
+	"github.com/stretchr/testify/require"
+)
+
+// assertAnchorTxPreCommitOut checks that the anchor transaction for the
+// minted asset includes a pre-commitment output for the supply commitment.
+// If an expected delegation key is provided, it verifies that it matches
+// the one used in the pre-commitment output. The function returns the
+// delegation key found in the asset metadata.
+func assertAnchorTxPreCommitOut(
+	t *harnessTest, tapd *tapdHarness, rpcAsset *taprpc.Asset,
+	expectedDelegationKey fn.Option[btcec.PublicKey]) btcec.PublicKey {
+
+	// Fetch metadata for the minted asset.
+	ctxb := context.Background()
+
+	metaResp, err := tapd.FetchAssetMeta(
+		ctxb, &taprpc.FetchAssetMetaRequest{
+			Asset: &taprpc.FetchAssetMetaRequest_AssetId{
+				AssetId: rpcAsset.AssetGenesis.AssetId,
+			},
+		},
+	)
+	require.NoError(t.t, err)
+
+	delegationKey, err := btcec.ParsePubKey(metaResp.DelegationKey)
+	require.NoError(t.t, err)
+
+	// If a specific delegation key is expected, verify it matches the one
+	// retrieved from the asset metadata.
+	expectedDelegationKey.WhenSome(func(expectedKey btcec.PublicKey) {
+		require.True(t.t, expectedKey.IsEqual(delegationKey))
+	})
+
+	// Parse anchor tx and confirm that one output is a supply commitment
+	// pre-commitment output.
+	var msgTx wire.MsgTx
+	err = msgTx.Deserialize(
+		bytes.NewReader(rpcAsset.ChainAnchor.AnchorTx),
+	)
+	require.NoError(t.t, err)
+
+	expectedTxOut, err := tapgarden.PreCommitTxOut(*delegationKey)
+	require.NoError(t.t, err)
+
+	// The pre-commitment output should be present in the anchor tx exactly
+	// once.
+	foundOnce := false
+	for idx := range msgTx.TxOut {
+		txOut := msgTx.TxOut[idx]
+		if txOut.Value != expectedTxOut.Value {
+			continue
+		}
+		if !bytes.Equal(txOut.PkScript, expectedTxOut.PkScript) {
+			continue
+		}
+
+		// We found a pre-commitment output, but it should only be
+		// present once.
+		if foundOnce {
+			t.t.Fatalf("found pre-commitment output more than once")
+		}
+
+		foundOnce = true
+	}
+	require.True(t.t, foundOnce)
+
+	return *delegationKey
+}
+
+// testPreCommitOutput tests that the pre-commitment output is correctly
+// included in the anchor transaction when minting an asset group with
+// universe/supply commitments enabled.
+func testPreCommitOutput(t *harnessTest) {
+	mintReq := issuableAssets[0]
```

Comment:
Use `mintReq := CopyRequest(issuableAssets[0])` here to make sure we don't influence the following tests?

Reply:
I know, these probably shouldn't have been pointers in the first place...